jbpm-wb-kie-server/jbpm-wb-kie-server-api/src/main/java/org/jbpm/workbench/ks/utils/KieServerUtils.java

```diff
@@ -119,17 +118,23 @@ public static KieServicesClient createKieServicesClient(final String endpoint,
     }
 
     public static CredentialsProvider getCredentialsProvider() {
-        CredentialsProvider credentialsProvider = new SubjectCredentialsProvider();
-        if (KIE_SERVER_KEYCLOAK) {
-            try {
-                credentialsProvider = new KeyCloakTokenCredentialsProvider();
-            } catch (UnsupportedOperationException e) {
-                LOGGER.warn(USE_KEYCLOAK_PROPERTY + " set to true, but keycloak libraries are not in the classpath", e);
+        return new CredentialsProvider() {
+
+            @Override
+            public String getHeaderName() {
+                return javax.ws.rs.core.HttpHeaders.AUTHORIZATION;
             }
-        }
-        LOGGER.debug("{} initialized for the client.",
-                     credentialsProvider.getClass().getName());
-        return credentialsProvider;
+
+            @Override
+            public String getAuthorization() {
+                Principal principal = SecurityIntegrationFilter.getRequest().getUserPrincipal();
+                if (principal instanceof KeycloakPrincipal) {
+                    return new KeyCloakTokenCredentialsProvider().getAuthorization();
+                } else {
+                    return new SubjectCredentialsProvider().getAuthorization();
```

Comment:
same thing here, couldn't we create this one time? If it is created for each request (didn't confirm if it is indeed created per request) I think it could lead to high memory usage.